### PR TITLE
Add `Bluetooth.availability` flow

### PR DESCRIPTION
**Pull request by twyatt**
Provides a flow that can be subscribed to for determining if bluetooth is available on the system:

```kotlin
Bluetooth.availability.collect {
    when (it) {
        Available -> // todo: handle when bluetooth is available
        is Unavailable -> println("Bluetooth is not available because ${it.reason}")
    }
}
```

Checks (and emits updates on changes) that the bluetooth adapter is on.

On Android 11 and earlier, performs an initial check for location services (which are no longer needed for Android 12 and newer). Unfortunately, could not find a way to monitor changes to location services status, so only the initial emission of status is provided.

On Android, bluetooth permissions are not taken into consideration for determining availability, as the mechanisms for checking/asking for permissions vary greatly.

Related to https://github.com/JuulLabs/kable/issues/288.

### CODE REVIEW

File: core/src/appleMain/kotlin/Bluetooth.kt
Comment (davidtaylor-juul) [inline core/src/appleMain/kotlin/Bluetooth.kt:32]: IMHO would be nice to surface `PoweredOff` as a state rather than `Unknown` for situations where the user has enabled Airplane mode so the UI can present advice accordingly.

Comment (twyatt) [reply core/src/appleMain/kotlin/Bluetooth.kt:32]: Unfortunately JS (https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth/onavailabilitychanged) only provides a boolean of availability, so it degrades the common API to the same available/unavailable state. 🤷

Comment (twyatt) [reply core/src/appleMain/kotlin/Bluetooth.kt:32]: Good call out. 👍

File: core/src/jsMain/kotlin/Bluetooth.kt
Comment (davidtaylor-juul) [inline core/src/jsMain/kotlin/Bluetooth.kt:27]: It looks like this will crash on the `checkNotNull` below if there is no bluetooth object, is that preferred to emitting `Unavailable`?

Also, if the browser does not support the `availabilitychanged` event this will never emit.

Perhaps it would be useful to emit `Unavailable` with a suitable reason for both these scenarios instead?

**Comment by donadev** [comment]
This feature will be very useful once available. Any prevision on when it will be merged to main?

**Comment by twyatt** [comment]
> This feature will be very useful once available. Any prevision on when it will be merged to main?

I'm hoping to dedicate some time to this in the coming days.

**Comment by twyatt** [comment]
Build is blocked by:

```
/Users/runner/work/kable/kable/core/src/androidMain/kotlin/Advertisement.kt:37: Error: Missing permissions required by BluetoothDevice.getName: android.permission.BLUETOOTH_CONNECT [MissingPermission]
        get() = bluetoothDevice.name
                ~~~~~~~~~~~~~~~~~~~~
```

Will be resolved by bumping/supporting Android target API to 31 (which I'll put up another PR for soon).

**Comment by burnhamd** [comment]
Looks good.  Does exactly what I was thinking.

**Comment by twyatt** [comment]
Note to self: integrate into SensorTag sample app before merging.

**Review by davidtaylor-juul — APPROVED** [review]

**Pull request merged**
